inference-engine/thirdparty/clDNN/src/include/loop_inst.h

```diff
@@ -106,6 +93,73 @@ struct typed_program_node<loop> : public typed_program_node_base<loop> {
         return (ndim - 1) - (axis - 2);
     }
 
+    // read scala value from data primitive
+    static int64_t read_scalar_value(memory::ptr mem, stream& stream) {
+        int64_t trip_count = 0;
+        const layout& prim_layout = mem->get_layout();
+
+        switch (prim_layout.data_type) {
+        case data_types::u8: {
+            mem_lock<uint8_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i8: {
+            mem_lock<int8_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i32: {
+            mem_lock<int32_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i64: {
+            mem_lock<int64_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        default:
+            assert(false);
```

Comment:
can we raise an exception instead of asserting?

Reply:
fixed it. thanks.